src/commands/sdk.ts

```diff
@@ -83,6 +84,54 @@ export class SdkCommands {
 				DartHoverProvider.clearPackageMapCaches();
 			}
 		}));
+		context.subscriptions.push(vs.commands.registerCommand("flutter.screenshot", async (uri) => {
+			let shouldNotify = false;
+			if (!uri || !(uri instanceof Uri)) {
+				// If there is no path for this session, get it from config
+				// If there is already a path set for this session, but the config differs
+				// set the session path to the config path and conrtinue execution. 
+				if (!this.flutterScreenshotPath) {
+					this.flutterScreenshotPath = config.flutterScreenshotPath;
+					shouldNotify = true;
+				} else if (config.flutterScreenshotPath && (this.flutterScreenshotPath !== config.flutterScreenshotPath)) {
+					this.flutterScreenshotPath = config.flutterScreenshotPath;
+					shouldNotify = true;
+				}
+
+				// If path is still empty it will bring up the folder selector.
+				if (!this.flutterScreenshotPath) {
+					const selectedFolder =
+						await window.showOpenDialog({ canSelectFolders: true, canSelectMany: false, openLabel: "Select folder to save screenshot" });
+					if (selectedFolder && selectedFolder.length > 0) {
+						// Set variable to selected path. This allows prompting the user only once.
+						this.flutterScreenshotPath = selectedFolder[0].path;
+						shouldNotify = true;
+					} else {
+						// Do nothing if the user cancelled the folder selection.
+						return;
+					}
+				}
+
+				// 	// if folder doesn't exist, create it
+				if (!fs.existsSync(this.flutterScreenshotPath)) {
```

Comment:
I think this might not be recursive? If you rebase on the screenshot branch, you should be able to call `mkDirRecursive`.